packages/patternfly-4/react-core/src/experimental/components/DataToolbar/DataToolbarContent.tsx

```diff
@@ -12,17 +13,107 @@ export interface DataToolbarContentProps extends React.HTMLProps<HTMLDivElement>
+  /** Flag indicating that the Clear all filters button should be visible */
+  showClearFiltersButton?: boolean;
 }
 
-export const DataToolbarContent: React.FunctionComponent<DataToolbarContentProps> = ({
-  className,
-  children,
-  breakpointMods = [] as DataToolbarBreakpointMod[],
-  ...props
-}: DataToolbarContentProps) => {
-  return (
-    <div className={css(styles.dataToolbarContent, formatBreakpointMods(breakpointMods, styles), className)} {...props}>
-      {children}
-    </div>
-  );
-};
+export interface DataToolbarContentState {
+  /** Flag indicating the if the expandable content's expanded state is consumer managed or not */
+  isConsumerManagedToggleGroup: boolean;
+  /** Flag indicating if the component managed state has expanded content or not */
+  componentManagedIsExpanded: boolean;
+}
+
+export class DataToolbarContent extends React.Component<DataToolbarContentProps, DataToolbarContentState> {
+  private expandableContentRef = React.createRef<HTMLDivElement>();
+
+  static defaultProps = {
+    isExpanded: false,
+    breakpointMods: [] as DataToolbarBreakpointMod[],
+    showClearFiltersButton: false
+  };
+
+  constructor(props: DataToolbarContentProps) {
+    super(props);
+
+    this.state = {
+      isConsumerManagedToggleGroup: props.isExpanded || !!props.toggleIsExpanded,
+      componentManagedIsExpanded: false
+    };
+  }
+
+  toggleIsExpanded = () => {
+    this.setState(prevState => ({
+      componentManagedIsExpanded: !prevState.componentManagedIsExpanded
+    }));
+  };
+
+  closeExpandableContent = () => {
+    this.setState(() => ({
+      componentManagedIsExpanded: false
+    }));
+  };
+
+  componentDidMount() {
+    const { isConsumerManagedToggleGroup } = this.state;
+
+    if (!isConsumerManagedToggleGroup) {
+      window.addEventListener('resize', this.closeExpandableContent);
+    }
+  }
+
+  componentWillUnmount() {
+    const { isConsumerManagedToggleGroup } = this.state;
+    if (isConsumerManagedToggleGroup) {
+      window.removeEventListener('resize', this.closeExpandableContent);
+    }
+  }
+
+  render() {
+    const {
+      className,
+      children,
+      isExpanded,
+      toggleIsExpanded,
+      id,
+      breakpointMods,
+      clearAllFilters,
+      showClearFiltersButton,
+      ...props
+    } = this.props;
+    const { isConsumerManagedToggleGroup, componentManagedIsExpanded } = this.state;
+
+    const expandableContentId = `${id}-expandable-content`;
+
+    return (
+      <div
+        className={css(styles.dataToolbarContent, formatBreakpointMods(breakpointMods, styles), className)}
+        {...props}
+      >
+        <DataToolbarContentContext.Provider
+          value={{
+            isExpanded: isConsumerManagedToggleGroup ? isExpanded : componentManagedIsExpanded,
+            toggleIsExpanded: isConsumerManagedToggleGroup ? toggleIsExpanded : this.toggleIsExpanded,
+            expandableContentRef: this.expandableContentRef,
+            expandableContentId
```

Comment:
It's difficult to understand the difference between isExpanded, toggleIsExpanded, expandableContent*, etc. It would be good to clarify these names.